### PR TITLE
feat: Adding netex files for multioperator flat fare tickets

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

Adding the netex files for the multi operator flat fare tickets.

## Testing instructions

Generate the netex file for the multioperator flat fare.

### CODE REVIEW

File: fdbt-dev/data/matchingData/multiOpFlatFareGeoZone.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    },
+    {
+      "day": "thursday",
+      "timeBands": [{ "startTime": "0800", "endTime": "1700" }]
+    }
+  ],
+  "ticketPeriod": {
+    "startDate": "2002-02-01T00:00:00.000Z"
+  },
+  "operatorName": "Test Operator",
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/matchingData/multiOpFlatFareGeoZone.json:23]: Operator name should be Blackpool Transport

File: fdbt-dev/data/matchingData/multiOpFlatFareMultiServices.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          "startTime": "",
+          "endTime": ""
+        }
+      ]
+    }
+  ],
+  "ticketPeriod": {
+    "startDate": "2002-02-01T00:00:00.000Z"
+  },
+  "operatorName": "Test Operator",
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/matchingData/multiOpFlatFareMultiServices.json:51]: operator name should be Blackpool Transport

File: fdbt-dev/data/netexData/multiOpFlatFareGeoZone.xml [hunk 3]
```diff
@@ -0,0 +1,1871 @@
...
+                <Street>Rigby Road, Blackpool FY1 5DD</Street>
+              </Address>
+              <PrimaryMode>bus</PrimaryMode>
+            </Operator>
+          </organisations>
+          <groupsOfOperators>
+            <GroupOfOperators version="1.0" id="operators@bus">
+              <Name>Bus Operators</Name>
+              <members>
+                <OperatorRef version="1.0" ref="noc:NWBT">Pilkingtonbus</OperatorRef>
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/netexData/multiOpFlatFareGeoZone.xml:143]: Where is LNUD?

File: fdbt-dev/data/matchingData/multiOpFlatFareGeoZone.json [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      "parentLocalityName": "Hebden Bridge",
+      "indicator": "at",
+      "street": "Midgley Road"
+    }
+  ],
+  "additionalNocs": [
+    "LNUD",
+    "NWBT"
+  ],
+  "operatorGroupId": 1,
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/matchingData/multiOpFlatFareGeoZone.json:225]: No ID's should be in this file

File: fdbt-dev/validate.log [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+> netex-output@1.0.0 netexConvert /Users/nitisharora/Projects/create-data/repos/fdbt-netex-output
+> NODE_ENV=development UNVALIDATED_NETEX_BUCKET=$UNVALIDATED_NETEX_BUCKET ts-node src/netex-convertor/run-local.ts "$EVENT_DATA"
+
+NeTEx generation starting for BLAC/carnetFlatFare.json, of type flatFare...
+Uploading file: BLAC/carnetFlatFare.xml
+NeTEx generation complete for type flatFare
+Starting validation of BLAC/carnetFlatFare.xml...
+Found '/Users/nitisharora/Projects/create-data/repos/fdbt-netex-output/.nvmrc' with version <14>
+Now using node v14.20.0 (npm v6.14.17)
+
```
Comment (RobCattonKPMG) [inline fdbt-dev/validate.log:21]: What is this file?

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts [hunk 6]
```diff
@@ -410,17 +411,28 @@ const getMultiServiceList = (
...
-                                                        product.productName
-                                                    }@${product.productDuration.replace(' ', '-')}`,
-                                                },
-                                            },
+                                            TimeIntervalPrice:
+                                                'productDuration' in product
+                                                    ? {
+                                                          version: '1.0',
+                                                          id: `op:${product.productName}@${salesOfferPackage.name}@service`,
+                                                          Amount: {
```
Comment (RobCattonKPMG) [inline repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:419]: Why are we only adding a TimeIntervalPrice when there is a productDuration now?

File: fdbt-dev/data/netexData/multiOpFlatFareGeoZone.xml [hunk 7]
```diff
@@ -0,0 +1,1906 @@
...
+      <RequestDetail>AllWithXref</RequestDetail>
+    </RequestPolicy>
+  </PublicationRequest>
+  <dataObjects>
+    <CompositeFrame version="1.0" id="epd:UK:BLAC:CompositeFrame_UK_PI_NETWORK_FARE_OFFER:Pass@BLAC_products:op" dataSourceRef="op:src" responsibilitySetRef="op:tariffs">
+      <ValidBetween>
+        <FromDate>2002-02-01T00:00:00.000Z</FromDate>
+      </ValidBetween>
+      <Name>Fares for Blackpool Transport</Name>
+      <Description>Period ticket for Blackpool Transport</Description>
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/netexData/multiOpFlatFareGeoZone.xml:36]: Screenshot 2023-01-31 at 13 05 25

Screenshot above is from the netex generator - I dont think we should be saying 'Period ticket'

File: fdbt-dev/data/netexData/multiOpFlatFareMultiServices.xml [hunk 8]
```diff
@@ -0,0 +1,1908 @@
...
+      <RequestDetail>AllWithXref</RequestDetail>
+    </RequestPolicy>
+  </PublicationRequest>
+  <dataObjects>
+    <CompositeFrame version="1.0" id="epd:UK:BLAC:CompositeFrame_UK_PI_LINE_FARE_OFFER:Pass@BLAC_products:op" dataSourceRef="op:src" responsibilitySetRef="op:tariffs">
+      <ValidBetween>
+        <FromDate>2002-02-01T00:00:00.000Z</FromDate>
+      </ValidBetween>
+      <Name>Fares for Blackpool Transport</Name>
+      <Description>Period ticket for Blackpool Transport</Description>
```
Comment (RobCattonKPMG) [inline fdbt-dev/data/netexData/multiOpFlatFareMultiServices.xml:36]: Again, period ticket when it should be 'Flat fare ticket'

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts [hunk 9]
```diff
@@ -410,17 +411,28 @@ const getMultiServiceList = (
...
+                                                              $t: `${salesOfferPackage.price || product.productPrice}`,
+                                                          },
+                                                          TimeIntervalRef: {
+                                                              version: '1.0',
```
[the remark this answers is not in the document]
Comment (RobCattonKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:423]: Also from the netex generator

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts (hunk 6 above)
Comment (NitishAroraKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:419]: as the timeinterval is not applicable to flatFare multiOp ticket and not present also in the flatFareGeoZone.xml

**Pull request merged**